stdlib/Dates/src/rounding.jl

```diff
@@ -85,176 +85,176 @@ function Base.floor(dt::DateTime, p::TimePeriod)
 end
 
 """
-    floor(x::Period, precision::T) where T <: Union{TimePeriod, Week, Day} -> T
+    floor(precision::T, x::Period) where T <: Union{TimePeriod, Week, Day} -> T
 
 Round `x` down to the nearest multiple of `precision`. If `x` and `precision` are different
 subtypes of `Period`, the return value will have the same type as `precision`.
 
-For convenience, `precision` may be a type instead of a value: `floor(x, Dates.Hour)` is a
-shortcut for `floor(x, Dates.Hour(1))`.
+For convenience, `precision` may be a type instead of a value: `floor(Dates.Hour, x)` is a
+shortcut for `floor(Dates.Hour(1), x)`.
 
 ```jldoctest
-julia> floor(Day(16), Week)
+julia> floor(Week, Day(16))
 2 weeks
 
-julia> floor(Minute(44), Minute(15))
+julia> floor(Minute(15) ,Minute(44))
 30 minutes
```

Comment:
This is a breaking change which is not allowed in Julia 1.x.

We have to keep the original argument order here, but perhaps we should add a warning in the docstring indicating that argument order for values differs from that of types.

Reply:
It's for anything that did not error in 1.8. Because `floor(Minute(15), Minute(44)` returned a value in Julia 1.8, it must return that same value in all future Julia 1.x versions. This is also true for `round(Minute(15), Minute(44))`, but not true for `floor(Minute, Minute(44))`

Reply:
@LilithHafner Have reverted it back. Can you review the code?